### PR TITLE
APM-5256: Verify Api Key Flow Testing

### PR DESCRIPTION
**Pull request by fayez166**
## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: tests/test_enhanced_verify_api_key.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        app = _create_function_scoped_test_app
+        apikey = app["credentials"][0]["consumerKey"]
+
+        proxy_resp = requests.get(
+            url=f"{nhsd_apim_proxy_url}/enhanced-verify-api-key",
+            headers={"apikey": apikey},
+            timeout=60
+        )
+
+        assert proxy_resp.status_code == 403
```
Comment (acarriedev) [inline tests/test_enhanced_verify_api_key.py:68]: Might be good to test the expected response too

File: tests/test_enhanced_verify_api_key.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        proxy_resp = requests.get(
+            url=f"{nhsd_apim_proxy_url}/enhanced-verify-api-key",
+            headers={"apikey": apikey},
+            timeout=60
+        )
+
+        assert proxy_resp.status_code == 401
+
+    @pytest.mark.parametrize("apikey, expected_status", [
+        ("123", 401), ("abc", 401), ("123abc", 401)
```
Comment (acarriedev) [inline tests/test_enhanced_verify_api_key.py:105]: Would be good potentially to test an api key that is valid but for a different proxy/product?

**Review by acarriedev — CHANGES_REQUESTED** [review]

**Review by acarriedev — CHANGES_REQUESTED** [review]

File: tests/test_enhanced_verify_api_key.py (hunk 2 above)
Comment (fayez166) [reply tests/test_enhanced_verify_api_key.py:111]: Added the test_valid_api_key_incorrect_product test scenario

File: tests/test_enhanced_verify_api_key.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        }
+        url = (
+            f"https://api.enterprise.apigee.com/v1/organizations/nhsd-nonprod/developers/"
+            f"{email}/apps/{app_name}/keys/{apikey}"
+        )
+
+        access_token = getenv("APIGEE_ACCESS_TOKEN")
+
+        # POST apiProducts to app
+        update_proxy_resp = requests.post(
```
[the remark this answers is not in the document]
Comment (fayez166) [reply tests/test_enhanced_verify_api_key.py]: Have updated this.

**Review by acarriedev — APPROVED** [review]

**Pull request merged**